### PR TITLE
[BEAM-1758] Option to disable metrics reporting.

### PR DESCRIPTION
**Pull request by aviemzur**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by aviemzur** [comment]
Run Spark ValidatesRunner

**Comment by aviemzur** [comment]
Run Spark ValidatesRunner

**Comment by aviemzur** [comment]
R: @bjchambers 
CC: @iemejia

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptions.java [hunk 1]
```diff
@@ -349,4 +349,9 @@ public Long create(PipelineOptions options) {
       return NEXT_ID.getAndIncrement();
     }
   }
+
+  @Description("Enable/disable metrics reporting.")
```
Comment (bjchambers) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptions.java]: What about runners that provide their own metrics (eg., number of elements processed)? Should this disable those as well? Why would a user want to disable metrics reporting?

**Comment by bjchambers** [comment]
What is the goal of this change? Specifically, why would users want to disable metrics? If this is a performance consideration, there are two sources of (potential) performance pain:

1. Actually reporting metrics (which I believe this PR addresses)
2. Keeping the MetricsContainer in sync.

In general, if this is something we actually feel is useful, it seems like disabling both of the above as well as making require less runner-specific code, would be worthwhile.

**Comment by aviemzur** [comment]
A case could be made either way but I think we better dev-list this. I assigned the JIRA ticket back to @iemejia who created it and I'll close this PR for now. It'll be here for reference if we want to bring it back.

**Pull request closed without merging**